tests/lib/prepare.sh

```diff
@@ -25,16 +25,15 @@ update_core_snap_for_classic_reexec() {
 
     # Now unpack the core, inject the new snap-exec/snapctl into it
     unsquashfs "$snap"
-    cp /usr/lib/snapd/snap-exec squashfs-root/usr/lib/snapd/
-    cp /usr/bin/snapctl squashfs-root/usr/bin/
-    # also inject new version of snap-confine and snap-scard-ns
-    cp /usr/lib/snapd/snap-discard-ns squashfs-root/usr/lib/snapd/
-    cp /usr/lib/snapd/snap-confine squashfs-root/usr/lib/snapd/
-    # also add snap/snapd because we re-exec by default and want to test
-    # this version
-    cp /usr/lib/snapd/snapd squashfs-root/usr/lib/snapd/
-    cp /usr/lib/snapd/info squashfs-root/usr/lib/snapd/
-    cp /usr/bin/snap squashfs-root/usr/bin/snap
+
+    # clean the old snapd libexec binaries, just in case
+    rm squashfs-root/usr/lib/snapd/*
+    # and copy in the current ones
+    # (explicit --preserve=mode to get setuid bit)
+    cp --preserve=mode /usr/lib/snapd/* squashfs-root/usr/lib/snapd/
```

Comment:
cp -a might be the simplest thing here, also these bits will conflict with #3027